src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -1,38 +1,31 @@
 package roomescape.controller;
 
 import roomescape.domain.Reservation;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-import roomescape.dto.ReservationRequest;
+import roomescape.dto.ReservationCreateRequest;
 import roomescape.service.ReservationService;
 
 import java.net.URI;
 
 @RestController
 public class ReservationController {
 
-    private ReservationService reservationService;
+    private final ReservationService reservationService;
 
     public ReservationController(ReservationService reservationService) {
         this.reservationService = reservationService;
     }
 
     @PostMapping("/reservations")
-    public ResponseEntity createReservation(@RequestBody ReservationRequest reservationRequest) {
-        Reservation reservation = reservationService.createReservation(reservationRequest);
-        if (reservation == null) {
-            return ResponseEntity.unprocessableEntity().body("이미 예약된 시간입니다.");
-        }
+    public ResponseEntity createReservation(@RequestBody ReservationCreateRequest reservationCreateRequest) {
+        Reservation reservation = reservationService.createReservation(reservationCreateRequest);
         return ResponseEntity.created(URI.create("/reservations/" + reservation.getId())).build();
     }
 
     @GetMapping("/reservations/{id}")
     public ResponseEntity showReservation(@PathVariable Long id) {
         Reservation reservation = reservationService.showReservation(id);
```

Comment:
ReservationController 에서도 도메인을 그대로 응답하고 있는 부분이 있네요!
이런 부분들을 쉽게 발견하기 위해서 제네릭 타입을 선언해주시면 도움이 될 것 같아요 :)

```suggestion
    @GetMapping("/reservations/{id}")
    public ResponseEntity<ReservationResponse> showReservation(@PathVariable Long id) {
        Reservation reservation = reservationService.showReservation(id);
```